clientconn.go

```diff
@@ -199,6 +199,8 @@ func WithTimeout(d time.Duration) DialOption {
 }
 
 // WithDialer returns a DialOption that specifies a function to use for dialing network addresses.
+// If an error is returned by the dial function, gRPC checks its Temporary() method to decide if
+// re-dialing is necessary.
```

Comment:
gRPC internals check the Temporary() method of the returned error (if it is not nil). If there is no balancer installed and the error is temporary, grpc interanls redial. The dial is failed otherwise.